src/test/test_marshal.cpp

```diff
@@ -20,9 +21,14 @@ constexpr int SRC = 9; // Any source register number.
 constexpr int IMM = -1; // Any imm value.
 constexpr int INVALID_REGISTER = R10_STACK_POINTER + 1; // Not a valid register.
 
+typedef struct {
+    ebpf_inst inst;
+    bpf_conformance_groups_t groups;
+} ebpf_instruction_template_t;
```

Comment:
```suggestion
struct ebpf_instruction_template_t {
    ebpf_inst inst;
    bpf_conformance_groups_t groups;
};
```
It may be simpler to have dedicated template fields for each field of ebpf_inst. Just a thought.